test/integration/acceptance/set_account_detail_test.cpp

```diff
@@ -0,0 +1,156 @@
+               kUser2, kUser2Keypair.publicKey(), kRole2, perms)
+        .build()
+        .signAndAddSignature(kAdminKeypair)
+        .finish();
+  }
+
+  const interface::types::AccountDetailKeyType kKey = "key";
+  const interface::types::AccountDetailValueType kValue = "value";
+  const std::string kUser2 = "user2";
+  const std::string kUser2Id =
+      kUser2 + "@" + IntegrationTestFramework::kDefaultDomain;
+  const crypto::Keypair kUser2Keypair =
+      crypto::DefaultCryptoAlgorithmType::generateKeypair();
+};
+
+/**
+ * @given a user without can_set_detail permission
+ * @when execute tx with SetAccountDetail command aimed to the user
+ * @then there is the tx in proposal
+ */
+TEST_F(SetAccountDetail, Self) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .skipBlock()
+      .sendTx(complete(baseTx(kUserId)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .done();
+}
+
+/**
+ * @given a pair of users and first one without permissions
+ * @when the first one tries to use SetAccountDetail on the second
+ * @then there is the tx in proposal
+ */
+TEST_F(SetAccountDetail, WithoutNoPerm) {
+  auto second_user_tx = makeSecondUser();
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .skipBlock()
+      .sendTx(second_user_tx)
+      .skipProposal()
+      .checkBlock([](auto &block) {
+        ASSERT_EQ(block->transactions().size(), 1)
+            << "Cannot create second user account";
+      })
+      .sendTx(complete(baseTx(kUser2Id)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given a pair of users and first one with can_set_detail perm
+ * @when the first one tries to use SetAccountDetail on the second
+ * @then there is the tx in proposal
+ */
+TEST_F(SetAccountDetail, WithPerm) {
+  auto second_user_tx = makeSecondUser();
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms({permissions::can_set_detail}))
+      .skipProposal()
+      .skipBlock()
+      .sendTx(second_user_tx)
+      .skipProposal()
+      .checkBlock([](auto &block) {
+        ASSERT_EQ(block->transactions().size(), 1)
+            << "Cannot create second user account";
+      })
+      .sendTx(complete(baseTx(kUser2Id)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .done();
+}
+
+/**
+ * @given a pair of users
+ *        AND second has been granted can_set_my_detail from the first
+ * @when the first one tries to use SetAccountDetail on the second
+ * @then there is no tx in proposal
+ */
+TEST_F(SetAccountDetail, DISABLED_WithGrantablePerm) {
```

Comment:
Can we add a small comment here that will describe the trigger which enables the test?